### PR TITLE
Stats Revamp: Likes Details Screen

### LINKED ISSUE #18603: [iOS] Stats: Total Likes Details Screen
The Likes detail screen is a second level screen accessible by tapping on the Total Likes card within Insights.

iOS-Total Likes

It contains a week selector at the top to navigate back through previous weeks.

Cards:

- Posts: A list of 6 posts on the site with the most views over the given period (likely using https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/top-posts/)

### PR DESCRIPTION
**Pull request by sla8c**
This PR adds the Details (2nd level) Likes screen

Fixes #18603

Note:
 * View More disclosure on Details screen doesn't do anything yet + will be handled in future PR

Design
image

Implementation
image

To test:
1. Build and run + navigate to Stats Insights for a site
2. Likes Total card should not be available
3. Click around insights, period stats and ensure there are no crashes
4. Now enable both of the new stats feature flags and build and run again
        * statsNewAppearance
        * statsNewInsights
5. From the top level Stats Insights screen add the following cards: Likes Total, Followers Total, Comments Total if not already visible.  All 3 of these cards should look exactly the same as before this PR (note that I moved makeSparklineData into StatsTotalInsightCell so that the logic could be shared between top level insights and 2nd level insights so this is to test there is no unintended impact)
6. On the top level Stats Insights screen for cards Comments Total and Followers Total - tap the right chevron which should navigate you to the details screen + there should be no change in functionality compared to before this PR
7. On the top level Stats Insights screen scroll to Likes Total and tap the right chevron
8. You should see the Likes Total card followed by the Post and Pages card
9. Tap back and forth on the DatePicker at the top + Like Totals and Posts and Pages should update accordingly

## Regression Notes
1. Potential unintended areas of impact
Stats screens

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift [hunk 1]
```diff
@@ -576,40 +576,14 @@ private extension SiteStatsInsightsViewModel {
         let periodSummary = periodStore.getSummary()
         updateMostRecentChartData(periodSummary)
 
-        let sparklineData: [Int] = makeSparklineData(countKey: \StatsSummaryData.likesCount)
-        let likesData = SiteStatsInsightsViewModel.intervalData(periodSummary, summaryType: .likes)
-
-        return StatsTotalInsightsData(count: likesData.count, difference: likesData.difference, percentage: likesData.percentage, sparklineData: sparklineData)
+        return StatsTotalInsightsData.createTotalInsightsData(periodStore: periodStore, statsSummaryType: .likes)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift:579]: I think this method could be named `createLikesTotalInsightsData`, like we have `createCommentsTotal...` used for the method below.

File: WordPress/Classes/ViewRelated/Stats/Insights/StatsTotalInsightsCell.swift [hunk 2]
```diff
@@ -11,6 +11,50 @@ struct StatsTotalInsightsData {
...
+
+    public static func createTotalInsightsData(periodStore: StatsPeriodStore, statsSummaryType: StatsSummaryType) -> StatsTotalInsightsData {
+        let periodSummary = periodStore.getSummary()
+
+        guard let periodSummary = periodStore.getSummary() else {
+            return StatsTotalInsightsData(count: 0)
+        }
+
+        let splitSummaryTimeIntervalData = SiteStatsInsightsViewModel.splitStatsSummaryTimeIntervalData(periodSummary)
+        let sparklineData: [Int] = makeSparklineData(countKey: \StatsSummaryData.commentsCount, splitSummaryTimeIntervalData: splitSummaryTimeIntervalData)
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/Insights/StatsTotalInsightsCell.swift]: I think this one should be `likesCount` as it's used to create the likes data?

Comment (frosty) [reply WordPress/Classes/ViewRelated/Stats/Insights/StatsTotalInsightsCell.swift]: Or can we work it out based on the `statsSummaryType` passed in, then we could just call into this method from `createCommentsTotalInsightsData`?

**Review by frosty — COMMENTED** [review]
Looking good! I just added a couple of comments, I'm not sure if we're showing comments data instead of likes in the sparkline.

Also, is it expected that if I navigate back past 2 weeks in the detail view I no longer see a sparkline or comparison in the Likes card at the top of the detail view?

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift:579]: Thanks!  I was trying to do something generic (thanks for noticing comments data) + I've updated with the generic intent I was hoping for originally

File: WordPress/Classes/ViewRelated/Stats/Insights/StatsTotalInsightsCell.swift (hunk 2 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/Insights/StatsTotalInsightsCell.swift]: >Or can we work it out based on the statsSummaryType passed in, then we could just call into this method from createCommentsTotalInsightsData?

Yep that is what I was going for!  I've updated 🙂

**Review by frosty — APPROVED** [review]

**Pull request merged**